rfc/rfc-51/rfc-51.md

```diff
@@ -64,69 +65,72 @@ We follow the debezium output format: four columns as shown below
 
 Note: the illustration here ignores all the Hudi metadata columns like `_hoodie_commit_time` in `before` and `after` columns.
 
-## Goals
+## Design Goals
 
 1. Support row-level CDC records generation and persistence;
 2. Support both MOR and COW tables;
 3. Support all the write operations;
 4. Support Spark DataFrame/SQL/Streaming Query;
 
-## Implementation
+## Configurations
 
-### CDC Architecture
+| key                                                 | default  | description                                                                                                                                      |
+|-----------------------------------------------------|----------|--------------------------------------------------------------------------------------------------------------------------------------------------|
+| hoodie.table.cdc.enabled                            | `false`  | The master switch of the CDC features. If `true`, writers and readers will respect CDC configurations and behave accordingly.                    |
+| hoodie.table.cdc.supplemental.logging               | `false`  | If `true`, persist the required information about the changed data, including `before`. If `false`, only `op` and record keys will be persisted. |
+| hoodie.table.cdc.supplemental.logging.include_after | `false`  | If `true`, persist `after` as well.                                                                                                              |
 
-![](arch.jpg)
+To perform CDC queries, users need to set `hoodie.table.cdc.enable=true` and `hoodie.datasource.query.type=incremental`.
 
-Note: Table operations like `Compact`, `Clean`, `Index` do not write/change any data. So we don't need to consider them in CDC scenario.
- 
-### Modifiying code paths
+| key                                    | default    | description                          |
+|----------------------------------------|------------|--------------------------------------|
+| hoodie.table.cdc.enabled               | `false`    | set to `true` for CDC queries        |
+| hoodie.datasource.query.type           | `snapshot` | set to `incremental` for CDC queries |
```

Comment:
see this: https://github.com/apache/hudi/pull/5885#issuecomment-1189748888 and the related discussion.
there are some differences between cdc query and inc query. i need to  another config like `hoodie.datasource.incremental.output` to distinguish both.

Reply:
agree that we should have an output format config and keep the query type as is. If they turn on cdc on writer side, they should change the incremental format accordingly.

Reply:
Shouldn't the cdc read just an incremental query here with a specific cdc format ? I mean the `hoodie.datasource.query.type` should force to be `incremental` if `hoodie.table.cdc.enabled` is true.

And i have a question: what is the behavior if user opens option `hoodie.table.cdc.enabled` and they want a snapshot query, should we allow that ?

Reply:
All the existing features are still supported when `hoodie.table.cdc.enabled` is true.

Reply:
We should make the user story/use cases clear, roughly enable all the options make things a mess and user would feel confused easily.

Reply:
yes. as we discussed above, i think the use case of hudi cdc is simple.

users need to use hudi in cdc scene, just set `hoodie.table.cdc.enabled` when write, and set `hoodie.datasource.query.type=incremental` and `hoodie.datasource.incremental.output.format=cdc` when read. also this table can be queried in the snapshot mode, or other hudi cases.